backend/controllers/teleservices.js

```diff
@@ -0,0 +1,119 @@
+var situations = require('./situations');
+var jwt = require('jsonwebtoken');
+var moment = require('moment');
+moment.locale('fr');
+
+var fields = [{
+  label: 'votre date de naissance',
+  value: '1940-12-12',
+  formatter: function(value) { return moment(value).format('LL'); },
+  key: 'date_naissance_dem'
+}, {
+  label: 'votre situation familiale',
+  value: 0,
+  formatter: function() { return 'Célibataire'; },
+  key: 'situationfam_dem'
+}, {
+  label: 'vos salaires (net) sur les 12 derniers mois',
+  value: 800*12,
+  formatter: function(value) { return value.toString() + ' €'; },
+  key: 'salaire_dem'
+}, {
+  label: 'votre retraite (net) sur les 12 derniers mois',
+  value: 500*12,
+  formatter: function(value) { return value.toString() + ' €'; },
+  key: 'montantRetraite_dem'
+}, {
+  label: 'vos allocations sur les 12 derniers mois',
+  value: 200*12,
+  formatter: function(value) { return value.toString() + ' €'; },
+  key: 'allocations_dem'
+}];
+
+var teleservices = [{
+  name: 'loiret_APA_test',
+  fields: fields,
+  destination: {
+    label: 'du Loiret (test)',
+    url: 'https://reflexe45-test.loiret.fr/public/requestv2/accountless/teleprocedure_id/92/'
+  }
+}, {
+  name: 'localtest',
+  fields: fields,
+  destination: {
+    label: 'en local',
+    url: 'http://localhost:3000/prefill/'
+  }
+}, {
+  name: 'livetest',
+  fields: fields,
+  destination: {
+    label: 'en ligne',
+    url: 'http://test.mes-aides.gouv.fr/prefill/'
+  }
+}];
+
+var teleserviceMap = teleservices.reduce(function(obj, ts) {
+  obj[ts.name] = ts;
+  return obj;
+}, {});
+
+exports.generator = function(teleservice) {
+  return function(req, res) {
+    var payload = {
+      id: req.situation._id,
+      scope: teleservice.name,
+      exp: Math.floor(Date.now() / 1000) + (60 * 60), // Expires after one hour
+    };
+
+    var token = jwt.sign(payload, req.situation.token);
+    var local = process.NODE_ENV == 'production' ? 'https://mes-aides.gouv.fr' : 'http://localhost:9000';
+    var callback = local + '/api/situations/via/' + token;
+
+    return res.json({
+      fields: teleservice.fields.map(function(field) {
+        return Object.assign(field, { formattedValue: field.formatter ? field.formatter(field.value) : field.value });
+      }),
+      destination: {
+        label: teleservice.destination.label,
+        url: teleservice.destination.url + '?callback=' + callback,
+      }
+    });
+  };
+};
+
+var prod = process.NODE_ENV == 'production';
+function fail(res, msg) {
+  if (prod)
+    return res.sendStatus(404);
+
+  return res.status(404).send({ error: msg });
+}
+
+exports.teleservice = function(req, res, next, token) {
```

Comment:
I think we could use a middleware here instead of coding the authentication logic inside the controller
With [Passport](http://www.passportjs.org/) or with a [simple function](https://scotch.io/tutorials/route-middleware-to-check-if-a-user-is-authenticated-in-node-js)